Networking/Networking/Remote/Remote.swift

```
@@ -235,28 +234,7 @@ public class Remote: NSObject {
     /// - Parameter request: Request that should be performed.
     /// - Returns: The result from the JSON parsed response for the expected type.
     func enqueue<M: Mapper>(_ request: Request, mapper: M) async throws -> M.Output {
-        try await withCheckedThrowingContinuation { continuation in
-            network.responseData(for: request) { [weak self] (result: Swift.Result<Data, Error>) in
-                guard let self else { return }
-
-                switch result {
-                case .success(let data):
-                    do {
-                        let validator = request.responseDataValidator()
-                        try validator.validate(data: data)
-                        let parsed = try mapper.map(response: data)
-                        continuation.resume(returning: parsed)
-                    } catch {
-                        DDLogError("<> Mapping Error: \(error)")
-                        self.handleResponseError(error: error, for: request)
-                        self.handleDecodingError(error: error, for: request, entityName: "\(M.Output.self)")
-                        continuation.resume(throwing: error)
-                    }
-                case .failure(let error):
-                    continuation.resume(throwing: self.mapNetworkError(error: error, for: request))
-                }
-            }
-        }
+        try await enqueueWithResponseHeaders(request, mapper: mapper).data
```

We should probably get eyes on any part of the codebase that's using this `enqueue` function, just to make sure we handle it properly in the layers above. We could have been relying on the way errors can be swallowed by tasks like this (at least, I think they can – the Alamofire cancellation one, for instance...?)

Looks like the remotes are:
- Blaze
- GenerativeContent
- GiftCardStats
- GoogleListings
- JustInTimeMessages
- MetaData
- Orders
- Payment (in-app purchase, probably unused)
- ProductBundle
- ProductCategories
- Products
- ProductsReports
- ShippingMethods
- StoreOnboardingTasks
- WCPay
- WordpressSite
- WordpressTheme

---

I haven't found explicit places where cancelling was important. The networking layer itself is quite well covered by unit tests, although the test seem to cover more conventional happy and unhappy paths. I did smoke testing around these areas, but haven't jumped on any out of place errors.